### PR TITLE
[DURACOM-131] Show output files in separate lines and a refresh spinner if status is running

### PR DESCRIPTION
**Pull request by enea4science**
## References
* Fixes DSpace/DSpace#8425

## Description
Show output files in separate lines and add a loading spinner that shows the time when the next refresh will be done

https://user-images.githubusercontent.com/127771679/236172137-a2b80aa3-dd50-4244-aa5b-67abdde87b4c.mov

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [ESLint](https://eslint.org/) validation using `yarn lint`
- [x] My PR doesn't introduce circular dependencies (verified via `yarn check-circ-deps`)
- [x] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new libraries/dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR includes new features or configurations, I've provided basic technical documentation in the PR itself.
- [x] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

### CODE REVIEW

**Review by artlowel — CHANGES_REQUESTED** [review]
Thanks @enea4science!

It works, but I have some concerns about the way it is implemented

All you have to do to get something to refresh is to set it to stale (and keep reRequestOnStale on true, which is the default), and use get**All**CompletedRemoteData instead of get**First**…

So the polling part doesn't need ever need to re-set `this.processRD$`. It will auto-update whenever it gets set to stale

I'd also try and move the whole polling process to a service, that way you can implement it without platform checks, only provide it on the browser side, and inject it into this component with an `@Optional()`

It has the added benefit of keeping all the code related to polling out of the component, which makes things easier to test, maintain and override

Something like `this.processPollingService.startPollingProcessStatus(process: Process, interval: number)`, which could return an id, which you could then use in the onDestroy of the component to call a stopPolling method

**Comment by enea4science** [comment]
Hello @artlowel 
Thanks for the review. 

The reason why I chose this approach, is because by using `getAllCompletedRemoteData` I'd have to wait 15 minutes (default cache time) in order to get the new refreshed data. But also, I'd need to show the timer there, so the polling and the counter still would be needed. 

Also, the reason I've not separated it into another service, is because this was only a 2 hours task, and it would require several additional hours, so we can keep it in a separate task. 

Thanks.

**Comment by artlowel** [comment]
> The reason why I chose this approach, is because by using `getAllCompletedRemoteData` I'd have to wait 15 minutes (default cache time) in order to get the new refreshed data. But also, I'd need to show the timer there, so the polling and the counter still would be needed.

I meant that if you set something to stale manually before those 15 minutes are up it will be re-retrieved immediately, and automatically. So I think it's the way to implement polling with the least amount of code



> Also, the reason I've not separated it into another service, is because this was only a 2 hours task, and it would require several additional hours, so we can keep it in a separate task.

I agree, 2 hours is not enough time to do this properly

**Comment by enea4science** [comment]
> I meant that if you set something to stale manually before those 15 minutes are up it will be re-retrieved immediately, and automatically. So I think it's the way to implement polling with the least amount of code

Is there any place where I can read more about this? 

Should we merge this and open another tasks to refactor it later? What should be the next step?

**Comment by artlowel** [comment]
>Is there any place where I can read more about this?

There's no developer documentation I'm aware of, however It looks like polling for a process to complete was already implemented before by setting requests to stale in the CollectionSourceControlsComponent: https://github.com/DSpace/dspace-angular/blob/f7bea3eaa990e09022d71cb636471dfbd305db4d/src/app/collection-page/edit-collection-page/collection-source/collection-source-controls/collection-source-controls.component.ts#L104-L109

The fact that it's needed again is a another argument to move it to a service, rather than duplicating it

**Comment by atarix83** [comment]
@enea4science  @artlowel  @tdonohue 

i agree the polling procedure can be refactored in order to be used in different places, but it will require more time.
So i think we need to decide if to go for the refactoring or leave as is the code and open a follow-up ticket to address it later.

I think the refactoring will take about 8-10hours

@tdonohue 

What's your opinion on the matter?

**Comment by tdonohue** [comment]
@atarix83 and @artlowel :  Honestly, I see this discussion as (somewhat) similar to the one that occurred in #2068 ... in that other ticket we all agreed that larger changes/refactors to `context.model.ts` were necessary. But, those changes were very complex to implement in that smaller PR.

In this PR it sounds like there's agreement that polling should be split out into a separate service so that we can avoid duplicate code.  However, similarly, it's a more complex solution than the one presented here.

So, I don't have a strong opinion, but see two possible solutions:
* If 4Science has the time to do so, they can create the polling service immediately in this PR.  (As this isn't paid work, this decision is up to 4Science)
* If 4Science doesn't feel there is time enough to make that happen, then @atarix83 should create a follow-up ticket to describe the changes needed to create such a polling service (perhaps even linking the the areas of code that could be replaced).  Then, we should see if it's possible to move this PR forward more "as-is"... cleanup what we can, but realize there is some duplicate code being added.

Overall, I'm OK with the code duplication _temporarily_ if that's all we have time for in 7.6.  However, we should create a ticket for follow-up work, as we don't want to keep around code duplication forever.

**Comment by enea4science** [comment]
Hello @tdonohue 
I merged main into the PR branch.

**Comment by tdonohue** [comment]
@artlowel : This looks ready to re-review given that we'll move your previous recommendations to a separate ticket.  (I can create the ticket when I get to reviewing this, if @atarix83 doesn't create the ticket.)

File: src/app/process-page/detail/process-detail.component.ts [hunk 1]
```diff
@@ -115,6 +127,53 @@ export class ProcessDetailComponent implements OnInit {
     );
   }
 
+  refresh() {
```
Comment (tdonohue) [inline src/app/process-page/detail/process-detail.component.ts:130]: It'd be nice if we could add specs for the new refresh / timer logic in this component.

File: src/app/item-page/orcid-page/orcid-sync-settings/orcid-sync-settings.component.ts [hunk 2]
```diff
@@ -156,7 +156,8 @@ export class OrcidSyncSettingsComponent implements OnInit {
         }
       }),
     ).subscribe((remoteData: RemoteData<ResearcherProfile>) => {
-      if (remoteData.isSuccess) {
+      // hasSucceeded is true if the response is success or successStale
+      if (remoteData.hasSucceeded) {
```
Comment (tdonohue) [inline src/app/item-page/orcid-page/orcid-sync-settings/orcid-sync-settings.component.ts:160]: This seems unrelated to this PR.  But, it seems like an OK change.  It'd just be nice to document what this is fixing.  Perhaps add a note to this PR's description as to why we are making this change to the ORCID-sync-settings component?  Or maybe split this into a separate, small PR to document what bug this fixes?

**Review by tdonohue — COMMENTED** [review]
@enea4science and @atarix83 : Overall, I really like this small improvement.  It works well, and I verified it works both when the process fails or succeeds.

I'm basically +1, but I had a few inline questions/comments.  It'd be nice if it's possible to add basic specs to this new code.  And there seems to be an unrelated change in this PR which I don't understand.

**Review by artlowel — APPROVED** [review]
I'm fine with merging this if Tim's comments are addressed

File: src/app/process-page/detail/process-detail.component.ts (hunk 1 above)
Comment (enea4science) [reply src/app/process-page/detail/process-detail.component.ts:130]: Added the unit tests

**Comment by enea4science** [comment]
> I'm fine with merging this if Tim's comments are addressed

I addressed Tim's comments in my last commit, ready last round of review.
